src/contracts/pods/EigenPod.sol

```diff
@@ -534,26 +533,18 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
         BeaconChainProofs.verifyValidatorFields({
             beaconStateRoot: beaconStateRoot,
             validatorFields: validatorFields,
-            validatorFieldsProof: balanceUpdateProof.validatorFieldsProof,
-            validatorIndex: validatorIndex
-        });
-
-        // Verify passed-in validator balanceRoot against verified beaconStateRoot:
-        BeaconChainProofs.verifyValidatorBalance({
-            beaconStateRoot: beaconStateRoot,
-            balanceRoot: balanceUpdateProof.balanceRoot,
-            validatorBalanceProof: balanceUpdateProof.validatorBalanceProof,
+            validatorFieldsProof: validatorFieldsProof,
             validatorIndex: validatorIndex
         });
 
         // Done with proofs! Now update the validator's balance and send to the EigenPodManager if needed
 
         uint64 currentRestakedBalanceGwei = validatorInfo.restakedBalanceGwei;
-         uint64 newRestakedBalanceGwei;
-        if (validatorBalance > MAX_RESTAKED_BALANCE_GWEI_PER_VALIDATOR) {
+        uint64 newRestakedBalanceGwei;
+        if (validatorEffectiveBalanceGwei > MAX_RESTAKED_BALANCE_GWEI_PER_VALIDATOR) {
```

Comment:
we're setting MAX_RESTAKED_BALANCE_GWEI_PER_VALIDATOR to 32 ETH now and effective balance is also capped at 32 ETH, so we can remove the if-else logic and just set newRestakedBalanceGwei = validatorEffectiveBalanceGwei

Reply:
Is there any edge case if ETH increases the effective balance cap from 32 ETH? Should we keep this logic or would we adequately assess impacts/upgrade well in advance?

Reply:
Perhaps, though at that point I'd imagine EL does the same, if the caps are raised to 64, for .  I think we should remove the if/else, makes it cleaner and more readable

Reply:
Hmm...so we still need the `MAX_RESTAKED_BALANCE_GWEI_PER_VALIDATOR` on withdrawals, and would be using it there.
So an upgrade would still be necessary in the event of a hard fork.

Definitely feeling like we shouldn't rush to just remove this -- we had a similar discussion in another PR, but I would rather have consistent behavior across EigenLayer and then change behavior consistently across-the-board in EigenLayer with an upgrade following/coordinated with any hard fork, rather than have a hard fork switch ~half of EigenLayer's behavior -- simply removing this feels like a recipe for weird behavior to happen in the event of a hard fork.

Reply:
ah yeah good point, gives us some breathing room to do the upgrade.  Lets keep it for now then